cartographer/mapping/internal/3d/pose_graph/optimization_problem_3d.h

```diff
@@ -92,7 +92,8 @@ class OptimizationProblem3D {
   // Optimizes the global poses.
   void Solve(const std::vector<Constraint>& constraints,
              const std::set<int>& frozen_trajectories,
-             const std::map<std::string, LandmarkNode>& landmark_nodes);
+             const std::map<std::string, LandmarkNode>& landmark_nodes,
+             bool freeze_landmarks);
```

Comment:
As discussed you can infer the that of `freeze_landmarks` from the state of `frozen_trajectories`.